scripts/synthetix/facilicate_trade.js

```diff
@@ -0,0 +1,97 @@
+
+// TODO - put this somewhere more generic. Copied from dex-contracts
+const sendTxAndGetReturnValue = async function (method, ...args) {
+  const result = await method.call(...args)
+  await method(...args)
+  return result
+}
+
+// truffle uses network by name and SynthetixJS uses network by ID
+const networkMap = {
+  "mainnet": 1,
+  "rinkeby": 4,
+}
+
+const argv = require("yargs")
+  .option("spread", {
+    type: "float",
+    describe: "Percentage increase required for trade (fees not accounted)",
+    default: 0.2,
+  })
+  .version(false).argv
+
+module.exports = async callback => {
+  try {
+    const snxjs = new SynthetixJs({ networkId: networkMap[argv.network] })
+
+    const exchange = await getExchange(web3)
+    const account = (await web3.eth.getAccounts())[0]
+
+    // Both of these hardcoded tokens have 18 decimal places.
+    // We "trust" that this will always be the case
+    // Although it seems that synthetix has the authority to upgade their token
+    // which could mean deploying a new one with a different number of decimal places.
+    const sETHKey = await snxjs.sETH.currencyKey()
+    const sUSDKey = await snxjs.sUSD.currencyKey()
+
+    const sETHAddress = await snxjs.Synthetix.synths(sETHKey)
+    const sUSDAddress = await snxjs.Synthetix.synths(sUSDKey)
+    console.log("sETH Address", sETHAddress)
+    console.log("sUSD Address", sUSDAddress)
+    const referenceTokenId = (await exchange.tokenAddressToIdMap.call(sUSDAddress)).toNumber()
+    const etherTokenId = (await exchange.tokenAddressToIdMap.call(sETHAddress)).toNumber()
+
+    const tokenInfoPromises = fetchTokenInfoFromExchange(exchange, [referenceTokenId, etherTokenId])
+
+    const stableToken = await tokenInfoPromises[referenceTokenId]
+    const targetToken = await tokenInfoPromises[etherTokenId]
+
+    // Note that sUSD is always 1 with synthetix
+    const exchangeRate = await snxjs.ExchangeRates.rateForCurrency(sETHKey)
+    const formatedRate = snxjs.utils.formatEther(exchangeRate)
+    console.log("sETH Price", snxjs.utils.formatEther(exchangeRate))
+
+    const batch_index = (await exchange.getCurrentBatchId.call()).toNumber()
+
+    const buyTokens = [etherTokenId, referenceTokenId]
+    const sellTokens = [referenceTokenId, etherTokenId]
+
+    const [upperSellAmount, upperBuyAmount] = calculateBuyAndSellAmountsFromPrice(
+      formatedRate * (1 + argv.spread / 100),
+      stableToken,
+      targetToken
+    )
+    const [lowerBuyAmount, lowerSellAmount] = calculateBuyAndSellAmountsFromPrice(
+      formatedRate * (1 - argv.spread / 100),
+      stableToken,
+      targetToken
+    )
+
+    const buyAmounts = [lowerBuyAmount, upperBuyAmount]
+    const sellAmounts = [lowerSellAmount, upperSellAmount]
+
+    const validFroms = Array(2).fill(batch_index)
+    const validTos = Array(2).fill(batch_index + 1)
```

Comment:
Is this correct? I thought an order where validFrom == validTo is valid for exactly one batch.

Reply:
This may be true, but if it is not, then I will have to submit an issue to `dex-telegram-bot` stating that its validity report is incorrect.

Observe the following two orders (one placed at 28 minutes with one minute expiry and the other at the beginning of a batch with 4 minute expiry).
 
<img width="1173" alt="Screenshot 2020-04-07 at 9 48 36 AM" src="https://user-images.githubusercontent.com/11778116/78608412-f77f8f80-78b4-11ea-98e2-52e5f3e6db50.png">


Will take a look at the contract.

Reply:
Yes. Indeed, setting validity period to have equal validFrom and validTo mean that an order is valid for one batch (since the intervals are _invlusive_) as we can see here:

https://github.com/gnosis/dex-contracts/blob/b075120c985d4ad725ab8aaf6a325500907868ca/contracts/BatchExchange.sol#L802-L804

I think @anxolin and @alfetopito will be interested to know this and I will now report an issue in the telegram bot.

Reply:
Yes, that's an issue in the telegram bot. I think TheGraph already takes this into account. Good call to do the issue and PR for that 👍